[bench] Adding a cpp benchmark to compare performance of nnc with static and symbolic shapes

navahgar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72198
* __->__ #72197
* #72196

Benchmark results:
```
--------------------------------------------------------------------------
Benchmark                                Time             CPU   Iterations
--------------------------------------------------------------------------
KernelBench/Eager/32                  2007 ns         2007 ns       350630
KernelBench/Eager/64                  2050 ns         2050 ns       335326
KernelBench/Eager/512                 2196 ns         2196 ns       320889
KernelBench/Eager/2048                2853 ns         2853 ns       233259
KernelBench/StaticShapes/32            544 ns          544 ns      1303151
KernelBench/StaticShapes/64            559 ns          559 ns      1271121
KernelBench/StaticShapes/512           586 ns          586 ns      1149529
KernelBench/StaticShapes/2048          820 ns          820 ns       855455
KernelBench/SymbolicShapes/32          704 ns          704 ns      1008014
KernelBench/SymbolicShapes/64          730 ns          730 ns       981898
KernelBench/SymbolicShapes/512         727 ns          727 ns       893909
KernelBench/SymbolicShapes/2048       1065 ns         1064 ns       648401
```

Differential Revision: [D33951742](https://our.internmc.facebook.com/intern/diff/D33951742)

navahgar committed:
[bench] Adding a cpp benchmark to compare performance of nnc with static and symbolic shapes

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33951742).

huiguoo commented on benchmarks/cpp/tensorexpr/bench_kernels.cpp:
```diff
@@ -0,0 +1,91 @@
+    for (auto _ : state) {
+      auto a = at::rand({dim}, at::TensorOptions(at::kCPU).dtype(at::kFloat));
+      auto b = at::rand({dim}, at::TensorOptions(at::kCPU).dtype(at::kFloat));
+      std::vector<at::Tensor> inputs = {a, b};
```
Should we exclude the input preparation process from measurement as well? So static and symbolic are compared fairly.

navahgar replied:
Good point. Moved.

huiguoo commented on benchmarks/cpp/tensorexpr/bench_kernels.cpp:
```diff
@@ -0,0 +1,91 @@
+  GraphWithSymbolicShapes(state);
+}
+
+BENCHMARK_REGISTER_F(KernelBench, StaticShapes)->Args({1500});
```
Is `1500` extracted from a real use case? Are we interested in benchmarking other dims like from 32 to 2048?

huiguoo reviewed (approved):
LGTM! 
I have two other comments below.

eellison commented:
where do the results of the benchmark show up ?

navahgar committed:
Update on "[bench] Adding a cpp benchmark to compare performance of nnc with static and symbolic shapes"


Differential Revision: [D33951742](https://our.internmc.facebook.com/intern/diff/D33951742)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33951742).

navahgar commented:
> where do the results of the benchmark show up ?

Added eager as well. Updated the PR description with the results.
